### PR TITLE
Add UI Testing to Examples-iOS app

### PR DESCRIPTION
**Pull request by DRosadoYew**
## Changes in this pull request

Add basic UI Testing to Examples-iOS app.
It doesn't cover all the examples yet, but Xcode is reporting ~82% of coverage.

Updated travis.yml to run the UI Tests for only one of the matrix jobs - "OS=10.0,name=iPhone 6s", to try to keep the build times acceptable. Last job took ~3 min for the UI Testing section (being the whole build usually around 40 min).

Was feeling like playing a bit with UI Testing and took the chance to use it here, since @rnystrom [showed interest in having it](https://github.com/Instagram/IGListKit/pull/251#issuecomment-262375087).

## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,21 +20,21 @@ env:
    - EXAMPLE_SCHEME="IGListKitExamples"
 
    matrix:
-   - DESTINATION="OS=8.1,name=iPhone 6" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="NO" BUILD_EXAMPLE="YES" POD_LINT="YES"
-   - DESTINATION="OS=8.2,name=iPhone 6" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="NO" BUILD_EXAMPLE="NO"  POD_LINT="NO"
-   - DESTINATION="OS=8.3,name=iPhone 6" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="NO" BUILD_EXAMPLE="NO"  POD_LINT="NO"
-   - DESTINATION="OS=8.4,name=iPhone 6" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="NO" BUILD_EXAMPLE="NO"  POD_LINT="NO"
+   - DESTINATION="OS=8.1,name=iPhone 6" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="NO" BUILD_EXAMPLE="YES" RUN_EXAMPLE_TESTS="NO" POD_LINT="YES"
```
Comment (jessesquires) [inline .travis.yml]: let's rename these `RUN_UI_TESTS` 😄

File: Examples/Examples-iOS/IGListKitExamples-UITests/DemosViewControllerUITests.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        let elem = app.collectionViews.cells.staticTexts[title]
+        if !elem.exists {
+            app.collectionViews.element.swipeUp()
+        }
+        
+        XCTAssertTrue(elem.exists)
+        elem.tap()        
+        XCTAssertTrue(app.navigationBars[title].exists)
+    }
+}
```
Comment (jessesquires) [inline Examples/Examples-iOS/IGListKitExamples-UITests/DemosViewControllerUITests.swift]: 💯

File: Examples/Examples-iOS/IGListKitExamples.xcodeproj/xcshareddata/xcschemes/IGListKitTodayExample.xcscheme [hunk 3]
```diff
@@ -43,13 +43,23 @@
...
+               ReferencedContainer = "container:IGListKitExamples.xcodeproj">
+            </BuildableReference>
+         </TestableReference>
       </Testables>
       <MacroExpansion>
          <BuildableReference
             BuildableIdentifier = "primary"
-            BlueprintIdentifier = "986FB72E1DBBAD8600A65C18"
-            BuildableName = "IGListKitTodayExample.appex"
-            BlueprintName = "IGListKitTodayExample"
```
Comment (jessesquires) [inline Examples/Examples-iOS/IGListKitExamples.xcodeproj/xcshareddata/xcschemes/IGListKitTodayExample.xcscheme]: 🤔 Hm.. not sure about this?

Comment (DRosadoYew) [inline Examples/Examples-iOS/IGListKitExamples.xcodeproj/xcshareddata/xcschemes/IGListKitTodayExample.xcscheme]: @jessesquires Weird that only the scheme from Today extension has these changes.
I just created a clean UI Test target, branched from master, and the same thing happened. 
Both Message and Today extensions build, run, and launch the UI Tests successfully.
Do you have any suggestions?

Comment (jessesquires) [inline Examples/Examples-iOS/IGListKitExamples.xcodeproj/xcshareddata/xcschemes/IGListKitTodayExample.xcscheme]: 👍 let's leave it for now. we'll forward fix if anything goes wrong.

**Comment by jessesquires** [comment]
Looks like travis is having issues: https://www.traviscistatus.com/incidents/nq0sf0srvx8s

Once it's green, we'll merge.

**Comment by rnystrom** [comment]
Slick!

**Pull request closed without merging**